src/dodal/utils.py

```diff
@@ -120,41 +121,49 @@ def make_all_devices(
         **kwargs: Arguments passed on to every device.
 
     Returns:
-        Dict[str, Any]: A dictionary of device name and device
+        Tuple[Dict[str, AnyDevice], Dict[str, Exception]]: This represents a tuple containing two dictionaries:
+
+    A dictionary where the keys are device names and the values are devices.
+    A dictionary where the keys are device names and the values are exceptions.
     """
     if isinstance(module, str) or module is None:
         module = import_module(module or __name__)
     factories = collect_factories(module, include_skipped)
-    devices: dict[str, AnyDevice] = invoke_factories(factories, **kwargs)
+    devices: Tuple[Dict[str, AnyDevice], Dict[str, Exception]] = invoke_factories(
+        factories, **kwargs
+    )
 
     return devices
 
 
 def invoke_factories(
     factories: Mapping[str, AnyDeviceFactory],
     **kwargs,
-) -> Dict[str, AnyDevice]:
+) -> Tuple[Dict[str, AnyDevice], Dict[str, Exception]]:
```

Comment:
Could: I think it's worth a bit of a docstring here on how the two dicts relate to each other. e.g. that every device will be represented in one list, either the first if it succeeds or the second if it fails? I wonder if a type like `Dict[str, AnyDevice | Exception]` would make this more clear but I'm not too sure.

Reply:
I also thing a general `docstring` on what's going on with the leaves etc. would be nice

Reply:
Have added some docstrings, the other option which I'm tempted by is to return a dataclass rather than a tuple. Will do that if no one objects.

Reply:
Fine by me